app-typescript/components/IconPicker.tsx

```diff
@@ -52,25 +52,29 @@ export class IconPicker extends React.PureComponent<IProps, IState> {
         this.props.onChange(item.value);
     }
 
-    triggerTemplate = (props: any) => <Button
-        icon={this.props.value}
-        text={this.props.value}
-        onClick={(e) => { props.onClick(e); }}
-        iconOnly={true} />
+    triggerTemplate = (props: {onClick: (e: React.SyntheticEvent) => void; }) => (
+        <Button
+            icon={this.props.value}
+            text={this.props.value}
+            onClick={(e) => { props.onClick(e); }}
+            iconOnly={true}
+        />
+    )
 
-    itemTemplate = ({ item }: { item: IItem | null }) => item ?
-        (<>
+    itemTemplate = ({ item }: { item: IItem | null }) => item && (
+        <>
             <Icon name={item.value} />
             <span className="sd-text--normal sd-padding-t--1">
                 {item.label}
             </span>
-        </>) : null
+        </>
+    )
 
     render() {
         return (
             <SelectGrid
-                label={this.props.label || "Icon"}
-                filterPlaceholder={this.props.filterPlaceholder || "Search..."}
+                label={this.props.label ?? "Icon"}
+                filterPlaceholder={this.props.filterPlaceholder ?? "Search..."}
```

Comment:
we will need those translated I guess, maybe we should rather make those props mandatory

Reply:
what if we treat those as the default fallback that's not translated, and if we need translations we pass them from the respective usage? im thinking here for all other components as well

Reply:
we will need a mechanism to translate it within the framework probably, it wouldn't make much sense to redo it everywhere we use it. for now the fallback is ok

Reply:
exactly, that's why I'd leave it